avm/src/vm/http.rs

```diff
@@ -120,3 +120,181 @@ macro_rules! make_server {
+                          match res {
+                            Ok(_) => { /* Do nothing */ }
+                            Err(_) => { /* Also do nothing? */ }
+                          }
+                        }
+                        Err(ee) => eprintln!(
+                          "Tunnel failed to connect to downstream on port {}: {}",
+                          $dest_port, ee
+                        ),
+                      };
+                    });
+                  }
+                  Err(ee) => eprintln!("Tunnel failed to open the socket? {}", ee),
+                };
+              }
+            });
+            true
+          }
+          Err(ee) => {
+            eprintln!("TCP tunnel failed to listen on port {}: {}", port_num, ee);
+            false
+          }
+        }
+      }
+      crate::vm::http::HttpType::HTTPS(https) => {
+        use futures_util::StreamExt;
+        let port_num = https.port;
+        let addr = std::net::SocketAddr::from(([0, 0, 0, 0], port_num));
+        let tls_cfg = {
+          let certs = rustls::internal::pemfile::certs(&mut std::io::BufReader::new(
+            https.cert.as_str().as_bytes(),
+          ));
+          let certs = certs.expect("Failed to load certificate");
+          let key = {
+            let keys = rustls::internal::pemfile::pkcs8_private_keys(&mut std::io::BufReader::new(
+              https.priv_key.as_str().as_bytes(),
+            ));
+            let keys = keys.expect("Failed to load private key");
+            if keys.len() != 1 {
+              panic!("Expected a single private key");
+            }
+            keys[0].clone()
+          };
+          let mut cfg = rustls::ServerConfig::new(rustls::NoClientAuth::new());
+          cfg.set_single_cert(certs, key).unwrap();
+          let alpns: Vec<Vec<u8>> = std::env::var("AVM_ALPN")
+            .unwrap_or("http/1.1".to_string())
+            .split(",")
+            .map(|s| s.to_string().into_bytes())
+            .collect();
+          cfg.set_protocols(alpns.as_slice().as_ref());
+          std::sync::Arc::new(cfg)
+        };
+        let tcp = tokio::net::TcpListener::bind(&addr).await;
+        match tcp {
+          Ok(tcp) => {
+            tokio::spawn(async move {
+              let tls_acceptor = tokio_rustls::TlsAcceptor::from(tls_cfg);
+              let incoming_tls_stream = async_stream::stream! {
+                use crate::vm::opcode::REGION_VMS;
+                use rand::{thread_rng, Rng};
+                loop {
+                  let l = REGION_VMS.read().unwrap().len();
+                  let i = async move {
+                    let mut rng = thread_rng();
+                    rng.gen_range(0..=l)
+                  }
+                  .await;
+                  let accept = tcp.accept().await;
+                  if accept.is_err() { continue; }
+                  let (mut socket, source_addr) = accept.unwrap();
```

Comment:
nit: can you make this a match? this should work
```suggestion
                  let (mut socket, source_addr) = match tcp.accept().await {
                    Ok(accepted) => accepted,
                    Err(_) => continue,
                  };
```

Reply:
Are you sure that works? A `continue` inside of an assignment statement doesn't make any sense.

Reply:
Just compiled it and it worked, still seems bonkers to me.